sympy/core/backend.py

```diff
@@ -0,0 +1,23 @@
+import os
+USE_SYMENGINE = os.getenv('USE_SYMENGINE')
+
+if USE_SYMENGINE:
+    from symengine.sympy_compat import (Symbol, Integer, sympify, S,
+        SympifyError, exp, log, gamma, sqrt, I, E, pi, Matrix,
+        sin, cos, tan, cot, csc, sec, asin, acos, atan, acot, acsc, asec,
+        sinh, cosh, tanh, coth, asinh, acosh, atanh, acoth,
+        lambdify, symarray, diff, zeros, eye, diag, ones, zeros,
+        expand, Function, symbols, var, Add, Mul, Derivative)
```

Comment:
Just a heads up, you will likely run into this issue as this gets integrated with more of SymPy (things that use the wrong pattern like `type(x) is Add` instead of `isinstance(x, Add)` will not work) https://github.com/sympy/sympy/issues/6751